### ISSUE TITLE
Extension fails to detect clang cl mode compilers when launched from Miniconda3

### ISSUE THREAD

**Issue by xoviat** [microsoft/vscode-cmake-tools#1633]
cl mode compilers are not detected:

![image](https://user-images.githubusercontent.com/49173759/105942591-23772780-6025-11eb-80ae-d12fcb61395f.png)

even though they are present:

![image](https://user-images.githubusercontent.com/49173759/105942681-54eff300-6025-11eb-96b9-8835a7608d55.png)

note command line configuration works fine. It's only the extension that fails.

![image](https://user-images.githubusercontent.com/49173759/105942735-6fc26780-6025-11eb-8e7b-974d88aa70c9.png)

extension version 1.5.3

I can provide additional information if requested, if I know what to look for.

**Comment by andreeis** [comment]
Thank you for opening this report. Can you confirm for us if this happens only when you open from Miniconda3 or also from a regular windows environment? I am asking because this normally works and we don't have a Miniconda3 environment handy at this moment to try.
Can you run the command "CMake: Scan for Kits" from the pallette and share with us the content of the cmake-tools-kits.json? You can have VSCode automatically open it for you if you run the command "CMake: Edit User-Local CMake Kits".

**Comment by xoviat** [comment]
I haven't tried the regular windows environment yet, because I use clang from miniconda3. It shouldn't be different, because vscode inherits the environment variables from its parent process. Where is cmake-tools-kits.json located?

**Comment by xoviat** [comment]
(On a different PC with the same issue): Sorry for not fully reading your response.

```
[
  {
    "name": "Clang 11.0.1 (x86_64-pc-windows-msvc)",
    "compilers": {
      "C": "C:\\Users\\User\\miniconda3\\Library\\bin\\clang.exe",
      "CXX": "C:\\Users\\User\\miniconda3\\Library\\bin\\clang++.exe"
    }
  },
  {
    "name": "GCC 10.2.0",
    "compilers": {
      "C": "C:\\ProgramData\\chocolatey\\bin\\gcc.exe",
      "CXX": "C:\\ProgramData\\chocolatey\\bin\\g++.exe"
    }
  },
  {
    "name": "GCC for arm-none-eabi 10.2.1",
    "compilers": {
      "C": "C:\\Program Files (x86)\\GNU Arm Embedded Toolchain\\10 2020-q4-major\\bin\\arm-none-eabi-gcc.exe",
      "CXX": "C:\\Program Files (x86)\\GNU Arm Embedded Toolchain\\10 2020-q4-major\\bin\\arm-none-eabi-g++.exe"
    }
  },
  {
    "name": "GCC for x86_64-w64-mingw32 10.2.0",
    "compilers": {
      "C": "C:\\ProgramData\\chocolatey\\bin\\x86_64-w64-mingw32-gcc.exe",
      "CXX": "C:\\ProgramData\\chocolatey\\bin\\x86_64-w64-mingw32-g++.exe"
    }
  },
  {
    "name": "Visual Studio Build Tools 2017 Release - amd64",
    "visualStudio": "5be6253d",
    "visualStudioArchitecture": "x64",
    "preferredGenerator": {
      "name": "Visual Studio 15 2017",
      "platform": "x64",
      "toolset": "host=x64"
    }
  },
  {
    "name": "Visual Studio Build Tools 2017 Release - amd64_x86",
    "visualStudio": "5be6253d",
    "visualStudioArchitecture": "x64",
    "preferredGenerator": {
      "name": "Visual Studio 15 2017",
      "platform": "win32",
      "toolset": "host=x64"
    }
  },
  {
    "name": "Visual Studio Build Tools 2017 Release - x86",
    "visualStudio": "5be6253d",
    "visualStudioArchitecture": "x86",
    "preferredGenerator": {
      "name": "Visual Studio 15 2017",
      "platform": "win32",
      "toolset": "host=x86"
    }
  },
  {
    "name": "Visual Studio Build Tools 2017 Release - x86_amd64",
    "visualStudio": "5be6253d",
    "visualStudioArchitecture": "x86",
    "preferredGenerator": {
      "name": "Visual Studio 15 2017",
      "platform": "x64",
      "toolset": "host=x86"
    }
  }
]
```

**Comment by andreeis** [comment]
I assume there is a clang-cl.exe on your system at this location C:\\Users\\User\\miniconda3\\Library\\bin\\ ? 

Did you write the first kit manually or it was detected? I will double check, but from what I remember the extension looks in a set of predefined paths and miniconda is not one of them. I wonder how the extension even detected clang(++).exe.

Note that on my system the extension detects lots of clang-cl.exe from various installs of Visual Studio. I don't have exactly Build Tools 2017 installed but from other 2017 and 2019 VSes I get their own clang-cl automatically detected. Maybe your Build Tools 2017 doesn't have clang installed (double check in Program Files where it is installed and confirm). If it's not installed, go to the Visual Studio Installer and search for clang components, most probably you'll see some check boxes that you can click and then Update your Build Tools 2017 to have clang provided with it, after which Scan for Kits should detect that.

More info for us now:
set cmake.loggingLevel to "Debug", clean the CMake Tools output channel, run scan kits again and share us the output channel new content.

**Comment by xoviat** [comment]
It was detected. If it's checking paths, then it's not a surprise that it wouldn't find it. I just assumed it would because it found the no-cl version.


Here is the debug log: https://gist.github.com/xoviat/ac0f6c0dbd0150083d507b49c01d89b5

**Comment by andreeis** [comment]
I think we have everything to investigate this. If you can, it wouldn't hurt also to get the output of running the 4 clang binaries with -v argument in a terminal. The extension makes some decisions of keeping or dropping a kit based on that output. It's worth to double check.

**Comment by xoviat** [comment]
Here you are:

```
(base) PS C:\Users\User\workspace\OpenBLAS\build> clang -v
clang version 11.0.1
Target: x86_64-pc-windows-msvc
Thread model: posix
InstalledDir: C:\Users\User\miniconda3\Library\bin
(base) PS C:\Users\User\workspace\OpenBLAS\build> clang++ -v
clang version 11.0.1
Target: x86_64-pc-windows-msvc
Thread model: posix
InstalledDir: C:\Users\User\miniconda3\Library\bin
(base) PS C:\Users\User\workspace\OpenBLAS\build> clang-cl -v
clang version 11.0.1
Target: x86_64-pc-windows-msvc
Thread model: posix
InstalledDir: C:\Users\User\miniconda3\Library\bin
(base) PS C:\Users\User\workspace\OpenBLAS\build>
```

**Comment by bobbrow** [comment]
That looks like it could have been fixed by PR #1620 
If you install the `cmake-tools.vsix` from the build action here (when it finishes): https://github.com/microsoft/vscode-cmake-tools/actions/runs/518763080 using the `Install from VSIX` command in VS Code, you can check to see if that resolves the issue.

**Comment by xoviat** [comment]
Doesn't appear to resolve the issue.
